### ISSUE TITLE
How to support single quotes?

### ISSUE THREAD

**Issue by kaifuzi** [mastersign/Mastersign.Expressions#18]
I can use expression `len(trim(\"  a b c  \"))`, I can get correct result.
But because I need to put expression in xml file, and xml doesn't accept `\"`
Is it possible to support single quotes? So that we can put exrepssion `len(trim('  a b c  '))` in xml file.

**Comment by mastersign** [comment]
Double quotes in XML are encoded with the entity `&quot;`.

See <https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references> for the other entities required in XML to represent characters used by the XML syntax.

**Comment by mastersign** [comment]
To encode a double quote inside a string of the _Mastersign.Expressions_ language, you can use the backslash as escape: `len("This \"dish\" is cold!")`

In XML you would encode it like `<expression>len(&quot;This \&quot;dish\&quot; is cold!&quot;)</expression>`.

By the way, I thing single quotes must be encoded as `&apos;` in XML and are no solution for your problem. That is, because both quote styles are allowed for quoting the value of an attribute. Thereby, both quote characters are part of the XML syntax and must be encoded via entities.

**Comment by kaifuzi** [comment]
Yes, you're right, we can use escape `&quot;` instead of double quotes.
But in xml we can type single quotes directly, using escape  `&apos;` is not mandatory, it's friendly for user typing, that's why I want Mastersign.Expressions can support single quotes.

**Comment by kaifuzi** [comment]
To support single quotes, for now I just replace `'` by `"`:
`if (expression.Contains("'"))`
`{
    expression = expression.Replace("'", "\"");
} `
But it's not a safe soluiton, if Mastersign.Expressions can support single quotes, then much better.

**Comment by mastersign** [comment]
@kaifuzi please check out the currernt HEAD of the master branch, if this solution is acceptable for you.

**Comment by kaifuzi** [comment]
@mastersign I download and test, but it still doesn't support single quotes:
![image](https://user-images.githubusercontent.com/30330327/121035318-0c53fd80-c7ae-11eb-93fd-92df12be4067.png)

**Comment by mastersign** [comment]
You can not mix quoting styles. This would make the syntax unnecessary complicated. You choose one quote style, that is best for the context you wan't to use the expression in and stick with it. You have to set the quote style via the new `LanguageOptions`:

```csharp
var context = new EvaluationContext
{
    Options = new LanguageOptionsBuilder()
        .WithQuoteCharacter(QuoteStyle.SingleQuote)
        .Build(),
};
```

**Comment by kaifuzi** [comment]
I'm clear now, and it works well as you said. This solution is good for me.
I have a question about properties of Mastersign.Expressions, you changed the options style, right? `SetIgnoreCase` doesn't work anymore, in release 0.6.0 we have to use `LanguageOptions` to initialize options, right?

**Comment by mastersign** [comment]
Yes, this breaking change is the reason for the version jump to 0.6 instead just moving to 0.5.3.

**Comment by mastersign** [comment]
Is this issue resolved in 0.6.0?

**Comment by kaifuzi** [comment]
Yes, it works well. I closed this request.
